iron-dropdown.html

```diff
@@ -340,6 +340,49 @@
           } else {
             Polymer.IronOverlayBehaviorImpl._applyFocus.apply(this, arguments);
           }
+        },
+
+        /**
+         * An element related to the dropdown (the triggering element, or an
+         * element within the dropdown) has lost focus.
+         */
+        _activeElementBlurred: function(event) {
+          if (event.target._blurListener) {
+            // Stop listening to blur events on the target.
+            event.target.removeEventListener('blur', event.target._blurListener);
+            event.target._blurListener = null;
+          }
+          this._dropdownFocusChanged(false);
+        },
+
+        /**
+         * Return true if the focus is anywhere within the dropdown.
+         */
+        _focusWithinDropdown: function() {
+          if (this.shadowRoot) {
+            return this.shadowRoot.activeElement !== null;
+          } else {
+            var activeElement = this._manager.deepActiveElement;
+            return Polymer.dom(this).deepContains(activeElement);
+          }
+        },
+
+        /**
+         * Overridden from `IronOverlayBehavior`.
+         */
+        _onCaptureFocus: function(event) {
+          if (!this._focusWithinDropdown()) {
```

Comment:
That sounded good to me, but it doesn't seem to work for paper-dropdown-menu under Shady DOM.

Under Shady DOM, opening a paper-dropdown-menu triggers some focus magic in IronControlState._focusBlurHandler, the result of which is that it's the paper-menu-button raising the focus event, not the paper-listbox. So when the above `_onCaptureFocus` is invoked, `this` (the dropdown) isn't in the event path.

To me it seems like the problem lies in iron-control-state, but that `_focusBlurHandler` feels like black magic, and I'm reluctant to touch it without more background on why it exists and how it works.

Aside: the perf issue here goes away in Shadow DOM, as we can directly inspect `this.shadowRoot.activeElement`.